SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,33 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only undummy child node, it will return that child node, else it will return itself
+    func mergeNode() -> Node {
+        var unDummyNum = 0
```

Comment:
Can you rename this to something like `numDummyChildren`?

Reply:
Ohh wait. unDummy doesn't really make sense semantically. You should call it `numNonDummyChildren`. You can't stick 'un' in front of 'dummy', you would use 'non'. English is weird :)

Reply:
An even better name would be `nonDummyChildrenCount`